03week/towersOfHanoi.js

```diff
@@ -13,61 +13,191 @@ let stacks = {
   c: []
 };
 
+//Print the currenct stack elements
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+//This method validates weather the source and destination stacks are
+//valid. It can be only a,b or c.
+function validateStackEntries(startStack, endStack){
+ if((startStack === 'a') || (startStack === 'b') || (startStack === 'c')) 
+ {
+   if(startStack === endStack)
+     return false;
+    if((endStack === 'a') || (endStack === 'b') || (endStack === 'c'))
+     return true;
+   else
+     return false;
+ }
+ else
+   return false;  
 }
+//This function will perform the mobe from source stack to destination stack
+function movePiece(startStack,endStack) {
+  if((startStack === 'a') && (endStack === 'b'))
+  {
+    stacks.b.push(stacks.a.pop());
+  }
+  else if((startStack === 'a') && (endStack === 'c'))
+  {
+    stacks.c.push(stacks.a.pop());
+  }
+  else if((startStack === 'b') && (endStack === 'a'))
+  {
+    stacks.a.push(stacks.b.pop());
+  }
+  else if((startStack === 'b') && (endStack === 'c'))
+  {
+    stacks.c.push(stacks.b.pop());
+  }
+  else if((startStack === 'c') && (endStack === 'a'))
+  {
+    stacks.a.push(stacks.c.pop());
+  }
+  else if((startStack === 'c') && (endStack === 'b'))
+  {
+    stacks.b.push(stacks.c.pop());
+  }
 
-function isLegal() {
-  // Your code here
+}
 
+//This function will check if the move is legal or not.
+//A move is legal only when the top entry of destination stack is higher than
+//the top entry in the source stack
+//Also a move is illegal from an empty stack
+function isLegal(startStack,endStack) {
+  if((startStack === 'a') && (endStack === 'b'))
```

Comment:
for every stack and letter, you're checking if the stack is a,b, or c, twice, that's a lot of repetition that you can condense.